### PR TITLE
feat(side-nav): add disabled prop to items

### PR DESCRIPTION
**Pull request by gergelyke**

### CODE REVIEW

File: src/side-navigation/nav-item.js [hunk 1]
```diff
@@ -43,15 +43,19 @@ class NavItem extends React.Component<NavItemPropsT> {
...
+    const tabIndex = {
+      tabIndex: item.disabled ? -1 : undefined,
+    };
     return (
       <NavLink
         href={item.itemId}
+        {...tabIndex}
         {...sharedProps}
         {...linkProps}
         {...(item.itemId
```
Comment (nadiia) [inline src/side-navigation/nav-item.js:55]:
```suggestion
        {...(item.itemId && !item.disabled
```

File: src/side-navigation/nav-item.js [hunk 2]
```diff
@@ -43,15 +43,19 @@ class NavItem extends React.Component<NavItemPropsT> {
...
           ? {
-              onClick: this.handleClick,
-              onKeyDown: this.handleKeyDown,
+              onClick: item.disabled ? () => {} : this.handleClick,
```
Comment (nadiia) [inline src/side-navigation/nav-item.js:57]:
```suggestion
              onClick: this.handleClick,
```

File: src/side-navigation/nav-item.js [hunk 3]
```diff
@@ -43,15 +43,19 @@ class NavItem extends React.Component<NavItemPropsT> {
...
+              onKeyDown: item.disabled ? () => {} : this.handleKeyDown,
```
Comment (nadiia) [inline src/side-navigation/nav-item.js:58]:
```suggestion
              onKeyDown: this.handleKeyDown,
```

File: src/side-navigation/styled-components.js [hunk 4]
```diff
@@ -69,6 +70,16 @@ export const StyledNavItemElement = styled<SharedPropsT>('div', props => {
   const paddingSuffixDir: string =
     $theme.direction === 'rtl' ? 'paddingLeft' : 'paddingRight';
 
+  let cursor = $selectable ? 'pointer' : 'default';
+  let color = $active ? colors.primary : null;
+  let hoverColor = $selectable ? colors.primary : null;
+
+  if ($disabled) {
+    cursor = 'not-allowed';
```
Comment (tajo) [inline src/side-navigation/styled-components.js:78]: Personally I would keep it as `default`. `not-allowed` seems too aggressive and we don't use it for disabled elements?

File: src/side-navigation/styled-components.js [hunk 5]
```diff
@@ -78,15 +89,15 @@ export const StyledNavItemElement = styled<SharedPropsT>('div', props => {
     [borderWidthDir]: '4px',
     [borderStyleDir]: 'solid',
     [borderColorDir]: $active ? colors.primary : 'transparent',
-    color: $active ? colors.primary : null,
-    cursor: $selectable ? 'pointer' : 'default',
+    color: color,
```
Comment (tajo) [inline src/side-navigation/styled-components.js:92]:
```suggestion
    color,
```

nit

File: src/side-navigation/styled-components.js [hunk 6]
```diff
@@ -78,15 +89,15 @@ export const StyledNavItemElement = styled<SharedPropsT>('div', props => {
...
+    cursor: cursor,
```
Comment (tajo) [inline src/side-navigation/styled-components.js:93]:
```suggestion
    cursor,
```

File: src/side-navigation/styled-components.js (hunk 4 above)
Comment (gergelyke) [reply src/side-navigation/styled-components.js:78]: we've discussed this offline, but for prosperity's sake: turns out ,we use it in quite a few places, so we've decided to keep things as is https://github.com/uber/baseweb/search?q=not-allowed&unscoped_q=not-allowed

**Pull request merged**